src/backend/optimizer/prep/prepunion.c

```diff
@@ -2059,28 +2097,10 @@ adjust_appendrel_attrs_mutator(Node *node,
 	 */
 	if (IsA(node, SubPlan))
 	{
-		SubPlan *sp = (SubPlan *) node;
-
-		if (!sp->is_initplan)
-		{
-			PlannerInfo *root = context->root;
-			Plan *newsubplan = (Plan *) copyObject(planner_subplan_get_plan(root, sp));
-			PlannerInfo *newsubroot = makeNode(PlannerInfo);
-
-			memcpy(newsubroot, planner_subplan_get_root(root, sp), sizeof(PlannerInfo));
+		plan_tree_base_prefix new_context;
+		new_context.node = (Node*)context->root;
```

Comment:
I suggest to run [pgindent](https://github.com/arenadata/gpdb/blob/6.20.4_arenadata34/src/tools/pgindent/README) on this file to keep code formatting clean.
Run it on one file - it will probably add a lot of formatting changes to whole file. No need to commit them all, just keep your new part of code modified by pgindent.

Reply:
I've fixed formatting of this line